// @svmkit/pulumi-svmkit/account/transfer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Transfer = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
class Transfer extends pulumi.CustomResource {
    /**
     * Get an existing Transfer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new Transfer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Transfer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Transfer.__pulumiType;
    }
    /**
     * Create a Transfer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.amount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'amount'");
            }
            if ((!args || args.connection === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connection'");
            }
            if ((!args || args.recipientPubkey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'recipientPubkey'");
            }
            if ((!args || args.transactionOptions === undefined) && !opts.urn) {
                throw new Error("Missing required property 'transactionOptions'");
            }
            resourceInputs["allowUnfundedRecipient"] = args ? args.allowUnfundedRecipient : undefined;
            resourceInputs["amount"] = args ? args.amount : undefined;
            resourceInputs["connection"] = args ? (args.connection ? pulumi.output(args.connection).apply(inputs.ssh.connectionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["recipientPubkey"] = args ? args.recipientPubkey : undefined;
            resourceInputs["transactionOptions"] = args ? args.transactionOptions : undefined;
        }
        else {
            resourceInputs["allowUnfundedRecipient"] = undefined /*out*/;
            resourceInputs["amount"] = undefined /*out*/;
            resourceInputs["connection"] = undefined /*out*/;
            resourceInputs["recipientPubkey"] = undefined /*out*/;
            resourceInputs["transactionOptions"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Transfer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Transfer = Transfer;
/** @internal */
Transfer.__pulumiType = 'svmkit:account:Transfer';
