// @svmkit/pulumi-svmkit/account/voteAccount.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoteAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const inputs = require("../types/input");
const utilities = require("../utilities");
class VoteAccount extends pulumi.CustomResource {
    /**
     * Get an existing VoteAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new VoteAccount(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VoteAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VoteAccount.__pulumiType;
    }
    /**
     * Create a VoteAccount resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.connection === undefined) && !opts.urn) {
                throw new Error("Missing required property 'connection'");
            }
            if ((!args || args.keyPairs === undefined) && !opts.urn) {
                throw new Error("Missing required property 'keyPairs'");
            }
            resourceInputs["authVoterPubkey"] = args ? args.authVoterPubkey : undefined;
            resourceInputs["closeRecipientPubkey"] = args ? args.closeRecipientPubkey : undefined;
            resourceInputs["connection"] = args ? (args.connection ? pulumi.output(args.connection).apply(inputs.ssh.connectionArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["keyPairs"] = args ? args.keyPairs : undefined;
        }
        else {
            resourceInputs["authVoterPubkey"] = undefined /*out*/;
            resourceInputs["closeRecipientPubkey"] = undefined /*out*/;
            resourceInputs["connection"] = undefined /*out*/;
            resourceInputs["keyPairs"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VoteAccount.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VoteAccount = VoteAccount;
/** @internal */
VoteAccount.__pulumiType = 'svmkit:account:VoteAccount';
