// @svmkit/pulumi-svmkit/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.watchtower = exports.validator = exports.types = exports.tuner = exports.solana = exports.networkinfo = exports.genesis = exports.faucet = exports.explorer = exports.agave = exports.account = exports.Provider = exports.KeyPair = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
exports.KeyPair = null;
utilities.lazyLoad(exports, ["KeyPair"], () => require("./keyPair"));
exports.Provider = null;
utilities.lazyLoad(exports, ["Provider"], () => require("./provider"));
// Export sub-modules:
const account = require("./account");
exports.account = account;
const agave = require("./agave");
exports.agave = agave;
const explorer = require("./explorer");
exports.explorer = explorer;
const faucet = require("./faucet");
exports.faucet = faucet;
const genesis = require("./genesis");
exports.genesis = genesis;
const networkinfo = require("./networkinfo");
exports.networkinfo = networkinfo;
const solana = require("./solana");
exports.solana = solana;
const tuner = require("./tuner");
exports.tuner = tuner;
const types = require("./types");
exports.types = types;
const validator = require("./validator");
exports.validator = validator;
const watchtower = require("./watchtower");
exports.watchtower = watchtower;
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "svmkit:index:KeyPair":
                return new exports.KeyPair(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("svmkit", "index", _module);
pulumi.runtime.registerResourcePackage("svmkit", {
    version: utilities.getVersion(),
    constructProvider: (name, type, urn) => {
        if (type !== "pulumi:providers:svmkit") {
            throw new Error(`unknown provider type ${type}`);
        }
        return new exports.Provider(name, undefined, { urn });
    },
});
